test/recipes/65-test_cmp_server.t

```diff
@@ -9,18 +9,52 @@
 # https://www.openssl.org/source/license.html
 
 use strict;
-use OpenSSL::Test qw/:DEFAULT data_file/;
+use OpenSSL::Test qw/:DEFAULT data_file srctop_dir bldtop_file bldtop_dir srctop_file/;
+
 use OpenSSL::Test::Utils;
 
+BEGIN {
 setup("test_cmp_server");
+}
+
+use lib srctop_dir('Configurations');
+use lib bldtop_dir('.');
+use platform;
+
+my $no_fips = disabled('fips') || ($ENV{NO_FIPS} // 0);
+
+$ENV{OPENSSL_MODULES} = bldtop_dir("providers");
+$ENV{OPENSSL_CONF_INCLUDE} = bldtop_dir("providers");
 
 plan skip_all => "This test is not supported in a no-cmp build"
     if disabled("cmp");
 
 plan skip_all => "This test is not supported in a no-ec build"
     if disabled("ec");
 
-plan tests => 1;
+plan tests => 2
+              + ($no_fips ? 0 : 2); #fips install + fips test
 
 ok(run(test(["cmp_server_test",
-             data_file("CR_protected_PBM_1234.der")])));
+             data_file("CR_protected_PBM_1234.der"),
+             "none"])));
+
+ok(run(test(["cmp_server_test",
+             data_file("CR_protected_PBM_1234.der"),
+             "default",
+             srctop_file("test", "default.cnf")])));
+
+unless ($no_fips) {
+    ok(run(app(['openssl', 'fipsinstall',
+                '-out', bldtop_file('providers', 'fipsmodule.cnf'),
+                '-module', bldtop_file('providers', platform->dso('fips')),
+                '-provider_name', 'fips', '-mac_name', 'HMAC',
```

Comment:
A lot of these parameters are set by default now.. (The hexkey has a different default than 00 so this will cause a error).
The fix is to simplify this to

```
    ok(run(app(['openssl', 'fipsinstall',
                '-out', bldtop_file('providers', 'fipsmodule.cnf'),
                '-module', bldtop_file('providers', platform->dso('fips'))])),
       "fipsinstall");
```

Reply:
Very good hint - done.